tests/components/lastfm/test_sensor.py

```diff
@@ -1,14 +1,74 @@
 """Tests for the lastfm sensor."""
+from pylast import Track, WSError
 
-from pylast import Track
-
-from homeassistant.components.lastfm.const import DOMAIN, STATE_NOT_SCROBBLING
+from homeassistant.components.lastfm.const import (
+    ATTR_LAST_PLAYED,
+    ATTR_PLAY_COUNT,
+    ATTR_TOP_PLAYED,
+    CONF_USERS,
+    DOMAIN,
+    STATE_NOT_SCROBBLING,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_API_KEY, CONF_PLATFORM, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.setup import async_setup_component
 
-from . import CONF_DATA, MockNetwork, patch_fetch_user
+from . import API_KEY, CONF_DATA, USERNAME_1, MockNetwork, patch_fetch_user
 
 from tests.common import MockConfigEntry
 
+LEGACY_CONFIG = {
+    Platform.SENSOR: [
+        {CONF_PLATFORM: DOMAIN, CONF_API_KEY: API_KEY, CONF_USERS: [USERNAME_1]}
+    ]
+}
+
+
+async def test_legacy_migration(hass: HomeAssistant) -> None:
+    """Test migration from yaml to config flow."""
+    with patch_fetch_user(None):
+        assert await async_setup_component(hass, Platform.SENSOR, LEGACY_CONFIG)
+        await hass.async_block_till_done()
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+    assert entries[0].state is ConfigEntryState.LOADED
+    issue_registry = ir.async_get(hass)
+    assert len(issue_registry.issues) == 1
+
+
+async def test_user_unavailable(hass: HomeAssistant) -> None:
+    """Test update when user can't be fetched."""
+    entry = MockConfigEntry(domain=DOMAIN, data={}, options=CONF_DATA)
```

Comment:
Can you make this a fixture

Reply:
Do you maybe have some good example about how to use fixtures in a place where the mock needs to handle different situations? I just tried to give it a shot but I think I am very much over engineering this and making it more complicated than I think should be useful